server/options.go

```diff
@@ -0,0 +1,79 @@
+package server
+
+import (
+	"github.com/storacha-network/go-ucanto/core/invocation"
+	"github.com/storacha-network/go-ucanto/core/ipld"
+	"github.com/storacha-network/go-ucanto/core/result"
+	"github.com/storacha-network/go-ucanto/server/transaction"
+	"github.com/storacha-network/go-ucanto/transport"
+)
+
+// Option is an option configuring a ucanto server.
+type Option func(cfg *srvConfig) error
+
+type srvConfig struct {
+	codec                 transport.InboundCodec
+	service               map[string]ServiceMethod[ipld.Builder, ipld.Builder]
+	validateAuthorization AuthorizationValidatorFunc
+	canIssue              CanIssueFunc
+	catch                 ErrorHandlerFunc
+}
+
+func WithServiceMethod[O, X ipld.Builder](can string, handleFunc ServiceMethod[O, X]) Option {
+	return func(cfg *srvConfig) error {
+		cfg.service[can] = func(input invocation.Invocation, context InvocationContext) (transaction.Transaction[ipld.Builder, ipld.Builder], error) {
+			tx, err := handleFunc(input, context)
+			if err != nil {
+				return nil, err
+			}
+			out := result.MapResultR0(
```

Comment:
something worth investigating here -- if we immediately commit to an ipld.Builder, is this generic worth it? I guess it will be useful for compile errors internal to the handler and type specifity when testing the handler in isolation